src/azul/transformer.py

```diff
@@ -138,36 +131,46 @@ def translate_field(cls, value: AnyJSON, path: Tuple[str, ...] = (), forward: bo
         elif field_type is None:
             return value
         else:
-            raise ValueError(f'Field type not found for path {path}')
+            raise ValueError(f'Unknown field_type value {field_type}')
 
     @classmethod
-    def translate_fields(cls, source: AnyJSON, path: Tuple[str, ...] = (), forward: bool = True) -> AnyMutableJSON:
+    def translate_fields(cls,
+                         doc: AnyJSON,
+                         field_types: Union[FieldType, FieldTypes],
```

Comment:
The type hint looks odd. Under what circumstances can this be a `FieldType`? Maybe we should pull up the use `Union` to the definition of FieldTypes?

Reply:
> Under what circumstances can this be a `FieldType`? 

`translate_fields` is recursive and as it descends into the dict `doc` it likewise descends into the `field_types` structure to pass that to the recursive call. Eventually `doc` is not a dict or list and `field_types` should be of type `FieldType` when sent to `translate_field`.

> Maybe we should pull up the use `Union` to the definition of FieldTypes?

Possible, however this is the only case where this is an issue and it might be beneficial to keep `FieldTypes` describe a mapping while `FieldType` describes a singular type.

Reply:
Ok.